src/app/Fake.Rsync/Rsync.fs

```diff
@@ -0,0 +1,415 @@
+        | MaxDelete num -> "--max-delete=" + string num
+        | MaxSize size -> "--max-size=" + size
+        | MinSize size -> "--min-size=" + size
+        | Partial -> "--partial"
+        | PartialDir dir -> "--partial-dir=" + dir
+        | DelayUpdates -> "--delay-updates"
+        | PruneEmptyDirs -> "--prune-empty-dirs"
+        | NumericIds -> "--numeric-ids"
+        | Timeout time -> "--timeout=" + string time
+        | IgnoreTimes -> "--ignore-times"
+        | SizeOnly -> "--size-only"
+        | ModifyWindow num -> "--modify-window=" + num
+        | TempDir dir -> "--temp-dir" + dir
+        | Fuzzy -> "--fuzzy"
+        | CompareDest dir -> "--compare-dest=" + dir
+        | CopyDest dir -> "--copy-dest=" + dir
+        | LinkDest dir -> "--link-dest=" + dir
+        | Compress -> "--compress"
+        | CompressLevel level -> "--compress-level=" + string level
+        | CvsExclude -> "--csv-exclude"
+        | Filter rule -> "--filter=" + rule
+        | F -> "-F"
+        | Exclude pattern -> "--exclude=" + pattern
+        | ExcludeFrom file -> "--exclude-from=" + file
+        | Include pattern -> "--include=" + pattern
+        | IncludeFrom file -> "--include-from=" + file
+        | FilesFrom file -> "--files-from=" + file
+        | From0 -> "--from0"
+        | Address address -> "--address=" + address
+        | Port port -> "--port=" + string port
+        | Sockopts options -> "--sockopts=" + options
+        | BlockingIO -> "--blocking-io"
+        | Stats -> "--stats"
+        | HeightBitsOutput -> "--8-bit-output"
+        | HumanReadable -> "--human-readable"
+        | Progress -> "--progress"
+        | P -> "-P"
+        | ItemizeChanges -> "--itemize-changes"
+        | OutFormat format -> "--out-format=" + format
+        | LogFile file -> "--log-file=" + file
+        | LogFileFormat format -> "--log-file-format=" + format
+        | PasswordFile file -> "--password-file=" + file
+        | ListOnly -> "--list-only"
+        | Bwlimit kbps -> "--bwlimit=" + kbps
+        | WriteBatch file -> "--write-bratch=" + file
+        | OnlyWriteBatch file -> "--only-write-batch=" + file
+        | ReadBatch file -> "--read-batch=" + file
+        | Protocol num -> "--protocol=" + string num
+        | ExtendedAttributes -> "--extended-attributes"
+        | Cache -> "--cache"
+        | Ipv4 -> "--ipv4"
+        | Ipv6 -> "--ipv6"
+        | Version -> "--version"
+        | Help -> "--help"
+
+    let private buildCommonArgs (param: Options) =
+        (param.Actions |> List.map actionToString)
+        @ param.Sources
+        @ [ param.Destination ]
+        |> String.concat " "
+
+    let exec (buildOptions: Options -> Options) args =
+        let results = new System.Collections.Generic.List<Fake.Core.ConsoleMessage>()
+        let timeout = TimeSpan.MaxValue
+
+        let errorF msg =
+            Trace.traceError msg
+            results.Add (ConsoleMessage.CreateError msg)
+
+        let messageF msg =
```

Comment:
I always thought we have a helper collecting the messages for you already. But I guess you wanted to have "live"-output?

Reply:
Yes ,I think having them live make more sense. And also, I believe I did that to get the right output color too.